radicle-cli/src/commands/review.rs

```diff
@@ -0,0 +1,205 @@
+                            anyhow!("invalid revision number `{}`", value.to_string_lossy())
+                        })?;
+                    revision = Some(id);
+                }
+                Long("sync") => {
+                    sync = true;
+                }
+                Long("no-sync") => {
+                    sync = false;
+                }
+                Long("message") | Short('m') => {
+                    let txt: String = parser.value()?.to_string_lossy().into();
+                    message.append(&txt);
+                }
+                Long("no-message") => {
+                    message = Comment::Blank;
+                }
+                Long("verbose") | Short('v') => {
+                    verbose = true;
+                }
+                Long("accept") if verdict.is_none() => {
+                    verdict = Some(Verdict::Accept);
+                }
+                Long("reject") if verdict.is_none() => {
+                    verdict = Some(Verdict::Reject);
+                }
+                Value(val) => {
+                    let val = val
+                        .to_str()
+                        .ok_or_else(|| anyhow!("patch id specified is not UTF-8"))?;
+
+                    id = Some(
+                        PatchId::from_str(val)
+                            .map_err(|_| anyhow!("invalid patch id '{}'", val))?,
+                    );
+                }
+                _ => return Err(anyhow::anyhow!(arg.unexpected())),
+            }
+        }
+
+        Ok((
+            Options {
+                id: id.ok_or_else(|| anyhow!("a patch id to review must be provided"))?,
+                message,
+                sync,
+                revision,
+                verbose,
+                verdict,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options, ctx: impl term::Context) -> anyhow::Result<()> {
+    let (_, id) =
+        rad::cwd().map_err(|_| anyhow!("this command must be run in the context of a project"))?;
+    let profile = ctx.profile()?;
+    let signer = term::signer(&profile)?;
+    let repository = profile.storage.repository(id)?;
+    let _project = repository
+        .project_of(profile.id())
+        .context(format!("couldn't load project {} from local state", id))?;
+    let cobs = cob::Store::open(*profile.id(), &repository)?;
+    let patches = cobs.patches();
+
+    let patch_id = options.id;
+    let patch = patches
+        .get(&patch_id)?
+        .context(format!("couldn't find patch {} locally", patch_id))?;
+    let patch_id_pretty = term::format::tertiary(term::format::cob(&patch_id));
+    let revision_ix = options.revision.unwrap_or_else(|| patch.version());
+    let _revision = patch
+        .revisions
+        .get(revision_ix)
+        .ok_or_else(|| anyhow!("revision R{} does not exist", revision_ix))?;
+    let message = options.message.get(REVIEW_HELP_MSG);
```

Comment:
The set of let statements are difficult to read.  What level of review would we like here?

I assume most of this are almost direct ports.  Might be best to review them for errors, and iterate after they are included?

Reply:
I think I could improve that by having multiple sections with comments?

Yes the code is copy-pasted and fixed to compile. I don't expect an indepth review, more like if something looks terribly wrong I'd like to know :)